src/components/CodeInput/PCodeInput.vue

```diff
@@ -12,49 +12,52 @@
 
     <template #control="{ attrs: ctrlAttrs }">
       <div ref="source" class="p-code-input__control" :class="classes.inputControl" @click.self="handleInputControlClick">
-        <div class="p-code-input__textarea-view-container">
-          <textarea
-            ref="textarea"
-            v-model="internalValue"
-            spellcheck="false"
-            class="p-code-input__textarea"
-            :rows="rows"
-            :class="classes.textArea"
+        <textarea
+          ref="textarea"
+          v-model="internalValue"
+          spellcheck="false"
+          class="p-code-input__textarea"
+          :placeholder="placeholder"
+          :rows="lines"
+          :class="classes.textArea"
+          :style="styles.textarea"
+          v-bind="ctrlAttrs"
+          @input="updateViewWidth"
+        />
+
+        <template v-if="lang">
+          <PCodeHighlight
+            :lang="lang"
+            :text="internalValue"
+            class="p-code-input__view"
+            :style="styles.view"
             v-bind="ctrlAttrs"
           />
+        </template>
 
-          <div class="p-code-input__view-container">
-            <template v-if="lang">
-              <PCodeHighlight
-                :lang="lang"
-                :text="internalValue"
-                class="p-code-input__view"
-                v-bind="ctrlAttrs"
-              />
-            </template>
-
-            <template v-else>
-              <PCode class="p-code-input__view" v-bind="ctrlAttrs">
-                {{ internalValue }}
-              </PCode>
-            </template>
-          </div>
-        </div>
+        <template v-else>
+          <PCode class="p-code-input__view" :style="styles.view" v-bind="ctrlAttrs">
+            {{ internalValue }}
+          </PCode>
+        </template>
       </div>
     </template>
   </p-base-input>
 </template>
 
 <script lang="ts" setup>
-  import { computed, ref } from 'vue'
+  import { useResizeObserver } from '@prefecthq/vue-compositions'
+  import { computed, onMounted, ref } from 'vue'
   import { PCode, PCodeHighlight, PLineNumbers } from '@/components'
   import { useScrollLinking } from '@/compositions'
   import { SupportedLanguage } from '@/types/codeHighlight'
 
   const props = defineProps<{
-    modelValue: string | null | undefined,
+    modelValue?: string | null,
```

Comment:
I don't think this change is correct. The modelValue is required. But you could bind an undefined value.